### PR TITLE
[MM-34527] fix reload and long names

### PR DESCRIPTION
**Pull request by Willyfrog**
Please provide the following information:

**Summary**

Pages were reloading after loading for the first time due to an exception being raised, which was treated as a load failure.
Also fixed long names getting in the way in several places.

**Issue link**

[MM-34527](https://mattermost.atlassian.net/browse/MM-34527)

**Screenshots**
Screen Shot 2021-03-30 at 17 25 38

Screen Shot 2021-03-31 at 15 42 27

### CODE REVIEW

File: src/main/views/MattermostView.js [hunk 1]
```diff
@@ -117,12 +118,12 @@ export class MattermostView extends EventEmitter {
     loadRetry = (loadURL, err) => {
         this.retryLoad = setTimeout(this.retry(loadURL), RELOAD_INTERVAL);
         WindowManager.sendToRenderer(LOAD_RETRY, this.server.name, Date.now() + RELOAD_INTERVAL, err.toString(), loadURL.toString());
-        log.info(`[${this.server.name}] failed loading ${loadURL}: ${err}, retrying in ${RELOAD_INTERVAL / SECOND} seconds`);
+        log.info(`[${Util.shorten(this.server.name)}] failed loading ${loadURL}: ${err}, retrying in ${RELOAD_INTERVAL / SECOND} seconds`);
```
Comment (devinbinnie) [inline src/main/views/MattermostView.js:121]: Is it as big of a deal to shorten the name in the console logs?

**Review by devinbinnie — COMMENTED** [review]
Looks good, where's the change to the page reload btw?

File: src/main/views/webContentEvents.js [hunk 2]
```diff
@@ -228,7 +228,6 @@ export const addWebContentsEventListeners = (mmview, getServersFunction) => {
     contents.on('page-title-updated', mmview.handleTitleUpdate);
     contents.on('page-favicon-updated', mmview.handleFaviconUpdate);
     contents.on('update-target-url', mmview.handleUpdateTarget);
-    contents.on(FOUND_IN_PAGE, mmview.handleFoundInPage);
```
Comment (hmhealey) [inline src/main/views/webContentEvents.js:231]: I assume this was the error that caused the refresh before. handleFoundInPage doesn't seem to be defined any more

**Review by hmhealey — APPROVED** [review]

File: src/main/views/MattermostView.js (hunk 1 above)
Comment (Willyfrog) [reply src/main/views/MattermostView.js:121]: not a big deal, but it was really annoying to have to chase the lines

File: src/main/views/webContentEvents.js (hunk 2 above)
Comment (Willyfrog) [reply src/main/views/webContentEvents.js:231]: yes, it is this one that was causing the problem. it was catched and retried loading, hence the flashing

**Review by matthewbirtch — APPROVED** [review]
Looks good to me as far as this PR goes and as far as I can tell from the UI.

@Willyfrog A couple things we should consider for future improvements below. Let me know if you'd like me to file tickets for them.
1. Add a maximum number of characters for Servers names
3. Consider a shorter max length for tab names before they start truncating. They feel a bit wide to me. 
2. When editing a server, if I only change the name but the URL doesn't change, we should just update the tab name rather than refreshing the whole server.

**Comment by jgilliam17** [comment]
@matthewbirtch We have an older ticket for limiting tab length/server name https://mattermost.atlassian.net/browse/MM-25464

**Comment by matthewbirtch** [comment]
> 2\. Consider a shorter max length for tab names before they start truncating. They feel a bit wide to me.
Ticket for this: https://mattermost.atlassian.net/browse/MM-34564



> 3\. When editing a server, if I only change the name but the URL doesn't change, we should just update the tab name rather than refreshing the whole server.
Ticket for this: https://mattermost.atlassian.net/browse/MM-34565

**Pull request merged**